experimental/segmenter/src/word.rs

```diff
@@ -298,9 +298,9 @@ where
     let mut s = String::new();
     s.push(left_codepoint);
     loop {
-        s.push(iter.current_pos_data.unwrap().1);
-        iter.current_pos_data = iter.iter.next();
-        if iter.current_pos_data.is_none() {
+        s.push(iter.get_current_codepoint());
+        iter.advance_iter();
+        if iter.is_eof() {
```

Comment:
Suggestion: a better pattern when using an iterator like this is to `if let Some(x) = y { ... } else { break }`.